### PR TITLE
Implement Availability Dashboard with Email Notifications

### PR DESCRIPTION
**Pull request by HanifZafari**
# Implement Availability Dashboard with Email Notifications

## Description

This PR introduces a comprehensive availability dashboard feature with both frontend and backend components. Key updates include:

1. Frontend and backend routes for the availability dashboard
2. Integration of environment variables for configuration
3. Email notification system using configurable SMTP (currently set to Gmail)
4. Secure link generation with encrypted IDs for email recipients
5. Auto-syncing dashboard mimicking Google Calendar functionality
6. New email sending endpoint for getting availablities `http://127.0.0.1:3000/sendInterviewEmails/{opening_id}`

The email system can be easily adapted to use different SMTP providers by modifying the environment variables. The dashboard's auto-sync feature eliminates the need for a manual save button.

### Key Features:
- **Availability Dashboard**: Implemented frontend and backend routes
- **Environment Variables**: Added for flexible configuration
- **Email Notifications**: 
  - Configurable SMTP (default: Gmail)
  - Secure links with encrypted IDs
  - New endpoint for sending availability emails
- **Auto-Sync**: Real-time updates without manual saving

### Testing Notes:
- Email sending endpoint created and can be tested manually via Postman
- To test, send a POST request to the new endpoint `http://127.0.0.1:3000/sendInterviewEmails/{opening_id}`
- There's a limit to the number of emails that can be sent
- For testing:
  1. Modify a row in the applications table:
     - Use your own email
     - Set an `opening_id` not used by other applicants
  2. Run the endpoint with the modified `opening_id`

### Notes:
- SMTP configuration can be changed by updating environment variables
- Emails are only sent to applications marked as 'applicant'
- UI integration (Not done): 'Email Availabilities' button in the view opening details page must call this endpoint with an opening_id

### CODE REVIEW

File: frontend/src/screens/AvailabilityCalendar.tsx [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+import parse from "date-fns/parse"; // Utility for parsing dates
+import startOfWeek from "date-fns/startOfWeek"; // Utility for determining the start of the week
+import getDay from "date-fns/getDay"; // Utility for getting the day of the week
+import "react-big-calendar/lib/css/react-big-calendar.css"; // Import base styles for the calendar
+import "react-big-calendar/lib/addons/dragAndDrop/styles.css"; // Import additional styles for drag-and-drop functionality
+import { enUS } from "date-fns/locale";
+import { useParams } from "react-router-dom";
+
+// Locale configuration for the calendar using date-fns
+const locales = { "en-US": enUS };
```
Comment (Rahul-Saxena-1) [inline frontend/src/screens/AvailabilityCalendar.tsx:16]: en-Au?

**Review by Rahul-Saxena-1 — APPROVED** [review]
Looks great!

**Review by FahadAssadi — COMMENTED** [review]
LGTM

**Review by FahadAssadi — APPROVED** [review]
LGTM

File: frontend/src/screens/AvailabilityCalendar.tsx (hunk 1 above)
Comment (HanifZafari) [reply frontend/src/screens/AvailabilityCalendar.tsx:16]: Changed it to enAu

**Pull request merged**